src/ConfigParser.cc

```diff
@@ -88,42 +89,25 @@ ConfigParser::strtokFile()
 
                 *t = '\0';
 
-                if ((wordFile = fopen(fn, "r")) == NULL) {
-                    debugs(3, DBG_CRITICAL, "ERROR: Can not open file " << fn << " for reading");
-                    return NULL;
-                }
-
-#if _SQUID_WINDOWS_
-                setmode(fileno(wordFile), O_TEXT);
-#endif
-
+                wordFile.reset(new Cfg::File(fn));
+                wordFile->load();
                 fromFile = 1;
             } else {
                 return t;
             }
         }
 
         /* fromFile */
-        if (fgets(buf, sizeof(buf), wordFile) == NULL) {
+        auto line = wordFile->nextLine();
+        if (line.isEmpty()) {
             /* stop reading from file */
-            fclose(wordFile);
-            wordFile = NULL;
+            wordFile = nullptr;
             fromFile = 0;
             return NULL;
         } else {
-            char *t2, *t3;
+            assert(line.length() < CONFIG_LINE_LIMIT);
+            SBufToCstring(buf, line);
             t = buf;
```

Comment:
The thing I'm worried about here is that many recipients of ```t``` do c-string parsing with arbitrary modifications of the buffer.

If any of that happens to step outside the buffer size, the char[] allows OS/compiler safety checks to catch the issue and abort. But with a MemBlob buffer it will overwrite/read into the surrounding objects or members in the pool allocation.

Reply:
Coming back to this, almost all callers will need a large re-write as they depend on this function return value having an implicit boolean conversion. This is too much bloat IMO. Leaving for a followup.

Reply:
> almost all callers will need a large re-write as they depend on this function return value having an implicit boolean conversion. This is too much bloat IMO

This change request should be addressed without changing the return type of this function. The provided solution sketch does not change the return type of this function.